tapgarden/caretaker.go

```diff
@@ -1164,6 +1185,39 @@ func SortSeedlings(seedlings []*Seedling) []string {
 	return allSeedlings
 }
 
+// SortAssets sorts the batch assets such that assets that are group anchors are
+// partitioned from all other assets.
+func SortAssets(fullAssets []*asset.Asset,
+	anchorVerifier proof.GroupAnchorVerifier) ([]*asset.Asset,
+	[]*asset.Asset) {
+
+	var anchorAssets, nonAnchorAssets []*asset.Asset
+	for ind := range fullAssets {
+		fullAsset := fullAssets[ind]
+
+		switch {
+		case fullAsset.GroupKey != nil:
+			err := anchorVerifier(
+				&fullAsset.Genesis,
+				fullAsset.GroupKey,
+			)
+
+			switch {
+			case err == nil:
+				anchorAssets = append(anchorAssets, fullAsset)
+			default:
```

Comment:
Should we instead return any errors that occur here? Otherwise they might be swallowed and only surface at a later place where they might be harder to pinpoint.

Reply:
Added concrete errors so we can now pass along anything else.